subsystem/kubeapi_test.go

```diff
@@ -0,0 +1,220 @@
+		ClusterNetworkHostPrefix: 23,
+		ServiceNetworkCidr:       "172.30.0.0/16",
+		IngressVip:               "1.2.3.9",
+		APIVip:                   "1.2.3.8",
+		SSHPublicKey:             sshPublicKey,
+		VIPDhcpAllocation:        false,
+		HTTPProxy:                "http://proxyserver:3218",
+		HTTPSProxy:               "http://proxyserver:3218",
+		NoProxy:                  "quay.io",
+		UserManagedNetworking:    false,
+		AdditionalNtpSource:      "test.ntp.source",
+		PullSecretRef:            secretRef,
+		ProvisionRequirements: v1alpha1.ProvisionRequirements{
+			ControlPlaneAgents: 3,
+			WorkerAgents:       0,
+			AgentSelector:      metav1.LabelSelector{MatchLabels: map[string]string{"key": "value"}},
+		},
+	}
+}
+
+func compareSpecToCluster(cluster *common.Cluster, spec *v1alpha1.ClusterSpec) {
+	Expect(cluster.Name).Should(Equal(spec.Name))
+	Expect(cluster.OpenshiftVersion).Should(Equal(spec.OpenshiftVersion))
+	Expect(cluster.BaseDNSDomain).Should(Equal(spec.BaseDNSDomain))
+	Expect(cluster.ClusterNetworkCidr).Should(Equal(spec.ClusterNetworkCidr))
+	Expect(cluster.ClusterNetworkHostPrefix).Should(Equal(spec.ClusterNetworkHostPrefix))
+	Expect(cluster.ServiceNetworkCidr).Should(Equal(spec.ServiceNetworkCidr))
+	Expect(cluster.IngressVip).Should(Equal(spec.IngressVip))
+	Expect(cluster.SSHPublicKey).Should(Equal(spec.SSHPublicKey))
+	Expect(swag.BoolValue(cluster.VipDhcpAllocation)).Should(Equal(spec.VIPDhcpAllocation))
+	Expect(cluster.HTTPProxy).Should(Equal(spec.HTTPProxy))
+	Expect(cluster.HTTPSProxy).Should(Equal(spec.HTTPSProxy))
+	Expect(cluster.NoProxy).Should(Equal(spec.NoProxy))
+	Expect(swag.BoolValue(cluster.UserManagedNetworking)).Should(Equal(spec.UserManagedNetworking))
+	Expect(cluster.OpenshiftVersion).Should(Equal(spec.OpenshiftVersion))
+	Expect(cluster.AdditionalNtpSource).Should(Equal(spec.AdditionalNtpSource))
+	Expect(cluster.PullSecret).Should(Equal(pullSecret))
+}
+
+func cleanUP(ctx context.Context, client k8sclient.Client) {
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Cluster{}, k8sclient.InNamespace(Options.Namespace))).To(BeNil())
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Image{}, k8sclient.InNamespace(Options.Namespace))).To(BeNil())
+}
+
+var _ = Describe("cluster installation", func() {
+	ctx := context.Background()
+
+	waitForClusterReconcileTimeout := 10
+
+	secretRef := &corev1.SecretReference{}
+
+	generateFAPostStepReply := func(h *models.Host, freeAddresses models.FreeNetworksAddresses) {
+		fa, err := json.Marshal(&freeAddresses)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = agentBMClient.Installer.PostStepReply(ctx, &installer.PostStepReplyParams{
+			ClusterID: h.ClusterID,
+			HostID:    *h.ID,
+			Reply: &models.StepReply{
+				ExitCode: 0,
+				Output:   string(fa),
+				StepID:   string(models.StepTypeFreeNetworkAddresses),
+				StepType: models.StepTypeFreeNetworkAddresses,
+			},
+		})
+		Expect(err).To(BeNil())
+	}
+
+	BeforeSuite(func() {
```

Comment:
http://assisted-jenkins.usersys.redhat.com/blue/organizations/jenkins/assisted-service/detail/PR-968/3/pipeline#step-37-log-19

Reply:
I think that if you use `Context()`, [same as done here](https://github.com/openshift/assisted-service/blob/2f49f536115b44b4a2270e78968818d36f8230bf/internal/host/hostcommands/install_cmd_test.go#L275), it should fix it.